python/lsst/dax/imgserv/image_v1.py

```diff
@@ -0,0 +1,228 @@
+#
+# LSST Data Management System
+# Copyright 2017 LSST/AURA.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+"""
+This module implements the Image ata model and use image getter object
+to retrieve the image per request specification.
+
+@author: Kenny Lo, SLAC
+
+"""
+
+class Image(object):
+    """ Image module maps a request and its parameters per JSON schema to the
+        corresponding imagegetter method.
+    """
+    def __init__(cls):
+       pass
+
+    @classmethod
+    def full_nearest(cls, image_getter, params):
+        """Get image nearest center.
+
+        Parameters
+        ----------
+        image_getter : getimage.imagegetter.ImageGetter
+        center.x, center.y, center.unit, filter
+
+        Returns
+        -------
+        lsst.afw.image
+            the full image (FITS).
+        """
+        x = float(params.get("center.x"))
+        y = float(params.get("center.y"))
+        unit = params.get("center.unit")
+        filt = params.get("nearest.filter")
+        image = image_getter.full_nearest(x, y, unit, filt)
+        return image
+
+    @classmethod
+    def full_from_data_id_rcff(cls, image_getter, params):
```

Comment:
`_rcff` is an opaque description for this method, and similarly with the `tpf` names, and non-obvious why it's named that based on the docstring. 

I'd potentially suggest `s/_rcff/_by_run/` and `s/_tpf/_by_tract/`, and in the docstring explain that the `params` object must have the additional information in it to validate.

It may also be more clear to not use the opaque `params` object, and perform input validation a layer up, I'm not sure.

Reply:
I'm going to remove 'rcff' and 'tp' tags altogether, by treating the 2 distinct cases as 'data_id' until it's needed to differentiate the two upon calling the butler.get().